Racket 8.6 broken on aarch64-darwin

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-env -iA nixpkgs.racket-minimal`

### Build log
```
compiler/cm:   start-compile: /private/tmp/nix-build-racket-minimal-8.6.drv-0/racket-8.6/src/mac/codesign.rkt
compiler/cm:   finish-compile: /private/tmp/nix-build-racket-minimal-8.6.drv-0/racket-8.6/src/mac/codesign.rkt
cs/c/racketcs -O info'@'compiler/cm -l- setup --boot ../setup-go.rkt cs/c/compiled ignored cs/c/ignored.d ../cs/known.rkt ../cs/
failed
 in build-one
 in loop
 in module->hash
make: *** [Makefile:13: all] Error 1
error: builder for '/nix/store/ilywy5nllnmjq0lz9xlbd1v7rzrg7axj-racket-minimal-8.6.drv' failed with exit code 2;
       last 10 log lines:
       > compiler/cm:   finish-compile: /private/tmp/nix-build-racket-minimal-8.6.drv-0/racket-8.6/src/cs/c/embed-boot.rkt
       > cs/c/raw_racketcs -O info'@'compiler/cm -l- setup --boot ../setup-go.rkt cs/c/compiled ignored cs/c/ignored.d ../mac/codesign.rkt -s - --entitlements ../mac/entitlements.plist cs/c/racketcs
       > compiler/cm:   start-compile: /private/tmp/nix-build-racket-minimal-8.6.drv-0/racket-8.6/src/mac/codesign.rkt
       > compiler/cm:   finish-compile: /private/tmp/nix-build-racket-minimal-8.6.drv-0/racket-8.6/src/mac/codesign.rkt
       > cs/c/racketcs -O info'@'compiler/cm -l- setup --boot ../setup-go.rkt cs/c/compiled ignored cs/c/ignored.d ../cs/known.rkt ../cs/
       > failed
       >  in build-one
       >  in loop
       >  in module->hash
       > make: *** [Makefile:13: all] Error 1
       For full logs, run 'nix log /nix/store/ilywy5nllnmjq0lz9xlbd1v7rzrg7axj-racket-minimal-8.6.drv'.
```

### Notify maintainers
@henrytill @vrthra

### Metadata
```
~ $ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.14 MiB unpacked):
  /nix/store/0gfg84zy171sgpwfbgwji3ybbjh38rh4-nix-info
  /nix/store/yarklbql3fs5b3r38h3w25nax2s8rnd3-DarwinTools-1
copying path '/nix/store/yarklbql3fs5b3r38h3w25nax2s8rnd3-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/0gfg84zy171sgpwfbgwji3ybbjh38rh4-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.8.0`
 - channels(hmih): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/interpreters/racket/force-remove-codesign-then-add.patch
+++ b/pkgs/development/interpreters/racket/force-remove-codesign-then-add.patch
@@ -5,6 +5,6 @@
  
  (void
 - (if remove?
-+ (if #t
++ (begin
       (remove-signature file)
       (add-ad-hoc-signature file)))

